imagej/imagej-ops2-tutorial/docs/CallingOps.md

```diff
@@ -0,0 +1,66 @@
+
+On this page, we will be constructing an `OpBuilder` call on an `OpEnvironment ops` to execute a Gaussian Blur on an input image `inImage`, with our output buffer `outImage`.
+
+## Specifying the name with `.op()`
+
+From the `OpEnvironment`, an `OpBuilder` call is initialized with the method `OpEnvironment.op(String)`, which is used to describe the name of the Op that the `OpBuilder` must return:
+
+```groovy
+ops.op("filter.gauss")
+```
+
+## Specifying the number of inputs with `.arity*()`
+
+After providing the name, the `OpBuilder` syntax requires the user to define the **number** of inputs to the Op call; we do this by adding one of the `arity*()` methods to the chain - `arity1()` tells the `OpBuilder` chain to expect one input, `arity2()` expects two inputs, and so on.
+
+In doing a basic gaussian blur, we will pass through the input image `inImage`, and a sigma parameter - therefore, we will call `arity2()`:
+
+```groovy
+ops.op("filter.gauss").arity2()
+```
+
+## Passing the inputs with `.input()`
+
+With the arity defined in the `OpBuilder` call, we can then chain the inputs with the `input()` method.
+
+For our gaussian blur, we will pass as inputs our input image `inImage`, and a `double` as our sigma parameter:
+
+```groovy
+ops.op("filter.gauss").arity2().input(inImage, 2.0)
+```
+
+## Passing an output buffer with `.output()`
+
+Now that the inputs are specified, we can chain the output buffer using the `output()` method.
+
+For our gaussian blur, we will pass as the output buffer our output image `outImage`:
+
+```groovy
+ops.op("filter.gauss").arity2().input(inImage, 2.0).output(outImage)
+```
+
+## Computing with `.compute()`
+
+With all of the components of the needed Op specified, we can begin computation with the `.compute()` method.
+
+```groovy
+ops.op("filter.gauss").arity2().input(inImage, 2.0).output(outImage).compute()
+```
+
+In the call to `compute()`, the `OpEnvironment` will use the components of the `OpBuilder` syntax to:
+* Match an Op based on the name provided, as well as the types of the provided input and output `Object`s
+* Execute the Op on the provided input and output `Object`s.
+
+## Additions: Repeating execution
+
+When an Op should be executed many times on different inputs, the `OpBuilder` syntax can be modified to return the *Op* instead. Instead of calling the `.compute()` function at the end of our `OpBuilder` call, we can instead call the `.computer()` method to get back the matched Op:
+
+If, instead of having a single image `inImage` we had Y images `inImage1`, `inImage2`, ..., `inImageY`, we 
+
+```groovy
+var gaussOp = ops.op("filter.gauss").arity2().input(inImage, 2.0).output(outImage).computer()
+gaussOp.compute(inImage)
```

Comment:
This does not compile for me, and it's not clear to me how it could even work.. shouldn't it be `gaussOp.compute(inImage, 2.0, outImage)`?

I think this section should be a modified to incrementally build up modifications to the original call:
1. the option to use `unary`, `binary`, etc.. in place of `ops.op(..).arity*()`
2. the option to use `inType` and `outType`
3. Using `Nil` with `inType` and `outType`
4. Using `compute()` and Op reuse
5. Model the `Function` and `Inplace` alternatives

Verify the code is working as expected in an IDE.

There should also be a `warnings` or `pitfalls` or something section. e.g. I pasted this into `OpsIntro` and it didn't compile when using `img` as the `inImage` because it's an `Img<?>`, and using a wildcard in the `input` method is a recipe for disaster I think.

Reply:
> This does not compile for me, and it's not clear to me how it could even work.. shouldn't it be `gaussOp.compute(inImage, 2.0, outImage)`?

Yup, good catch!

> 1. the option to use `unary`, `binary`, etc.. in place of `ops.op(..).arity*()`

Sure, I added that section in 86b6132

> 2. the option to use `inType` and `outType`

Man, I'm really hesitant to put that in what should be the most approachable page of the docs - using `Type`s for matching is *so* verbose and complex.

...honestly, what do we *really* use `inType`/`outType` patterns *for* nowadays? The only thing I think *might* be useful is `outType` for function matching - we don't use `inType`/`outType` in the ImageJ Ops2 tests *at all*.

@hinerm @ctrueden I think it would be really worthwhile to have a discussion about removing the `inType`/`outType` syntax from the OpBuilder (of course the `OpEnvironment` still needs the Type/`Nil` syntax) - we could potentially remove a lot the `OpBuilder` autocomplete clutter

Needless to say, I haven't added anything past 2 either.

> Verify the code is working as expected in an IDE.

The existing code compiles now

> There should also be a warnings or pitfalls or something section. e.g. I pasted this into OpsIntro and it didn't compile when using img as the inImage because it's an Img<?>, and using a wildcard in the input method is a recipe for disaster I think.

Good idea - added a `Common Pitfalls` section.